### PR TITLE
Noname 130 survey management

### PR DESCRIPTION
**Pull request by mtszlachetka**
Added option to delete surveys

### CODE REVIEW

File: backend/src/api/routes/surveys.py [hunk 1]
```diff
@@ -36,7 +37,9 @@ async def get_surveys_for_user(
         raise HTTPException(status_code=400, detail="User not found")
 
     user_id = user_crud.get_user_by_email(user.user_email, session).id
-    user_surveys = survey_crud.get_all_surveys_for_user(user_id, session)
+    user_surveys = survey_crud.get_not_deleted_surveys_for_user(
```
Comment (yanvoi) [inline backend/src/api/routes/surveys.py:40]: Do not create a new get_not_deleted_surveys_for_user method. Instead, reimplement get_all_surveys_for_user to only return non-deleted surveys. We will never want to return deleted surveys, and only the crud component should be aware of how the deletion is handled. The crud component should hide details like this.

**Review by yanvoi — CHANGES_REQUESTED** [review]

File: backend/src/db/crud/survey.py [hunk 2]
```diff
@@ -5,17 +5,41 @@
 
 
 def get_survey_by_code(survey_code: str, session: Session) -> Survey:
-    statement = select(Survey).where(Survey.survey_code == survey_code)
+    statement = select(Survey).where(
+        (Survey.survey_code == survey_code)
+        & (Survey.is_deleted == False)  # noqa: E712
```
Comment (sjerzykiewicz) [inline backend/src/db/crud/survey.py:10]: (Survey.is_deleted is False) should work (without noqa)

Comment (mtszlachetka) [reply backend/src/db/crud/survey.py:10]: Doesn't work, query returns no surveys, tried that before.

Comment (sjerzykiewicz) [reply backend/src/db/crud/survey.py:10]: ok, thats fine then

**Review by yanvoi — APPROVED** [review]

**Pull request merged**